### ISSUE TITLE
Proper mapping of PUF e18400 and e18500 to IRS-reported SALT aggregates

### ISSUE THREAD

**Issue by donboyd5** [PSLmodels/tax-microdata-benchmarking#118]
@nikhilwoodruff, FYI @martinholmer 

As mentioned on the phone today, I think tax-microdatabenchmarking currently maps the PUF SALT variable e18400 incorrectly to IRS-published total SALT rather than to the combined PIT-sales tax SALT component. However it is more complicated than I realized at the time; here are full details.

The PUF 2015 documentation (p.8) lists relevant variables as follows:
![image](https://github.com/PSLmodels/tax-microdata-benchmarking/assets/6538392/8bf40742-ec17-4df0-b93c-9b1e259ea62a)

On page A-7 it shows the variables on 2015 Schedule A:

![image](https://github.com/PSLmodels/tax-microdata-benchmarking/assets/6538392/a71918ed-cc1e-482f-972a-739acbb2471b)

**e18400 is combined PIT and sales tax, while e18500 is real estate taxes.**

On page 48 the PUF 2015 documentation gives us **ERRONEOUS NUMBERS FOR** (1) the official IRS totals for these variables in the column 2015 Full SOI Individual Sample ($35.270 billion and $18.861 billion), and (2) what we should expect for weighted sums in the 2015 PUF, including the 4 aggregate records in the column 2015 Public Use Sample (rounding to the same numbers as the full sample). These numbers are wrong and appear to be missing a digit.

![image](https://github.com/PSLmodels/tax-microdata-benchmarking/assets/6538392/66d8f4ac-a94a-4dc4-8d6e-cdbf29a6d9a6)

The correct numbers for the Full Sample are in the IRS file 15in14ar.xls, in BT10 and BY10, which are $352.701 billion and $188.606 billion (the PUF 2015 documentation appears to have several other missing-digit errors for other itemized deductions, so caveat emptor):

![image](https://github.com/PSLmodels/tax-microdata-benchmarking/assets/6538392/b0760780-1ab9-4f08-95af-d12ddd7de769)

This tells us the proper mapping and approximately what we should expect if we summarize the 2015 PUF and we don't exclude the 4 aggregate records.

Now that we know the proper mapping, we can worry about how to target the PUF in 2021. Here are the IRS published totals for 2021, from 21in14ar.xls:

![image](https://github.com/PSLmodels/tax-microdata-benchmarking/assets/6538392/283af83e-33e1-4712-92dc-e1e5efe87ee4)

So we should target 2021 values for filers for e18400 of $258.640 billion and for e18500, $99.984 billion.

I have updated my [IRS targets repo](https://github.com/donboyd5/irs_targets) to include the distributional detail. The data folder includes an updated potential_targets.csv file with separate variables id_pitgst (itemized deductions PIT and general sales tax), id_retax (real estate tax), and their number-of-returns counterparts nret_....

**Comment by donboyd5** [comment]
@nikhilwoodruff 

For completeness, here's the linkage between puf variables and the variable names I've given to IRS targets in the "potential_targets.csv" file:

- e18400 -- maps to id_pitgst (itemized deductions for personal income tax and general sales tax)
- e18500 -- maps to id_retax (itemized deductions for real estate tax)

And also (pseudocode):

- (e18400 > 0) * s006 -- maps to nret_id_pitgst
- (e18500 > 0) * s006 -- maps to nret_id_retax

**Comment by MaxGhenis** [comment]
Have you reported these inaccuracies to SOI?

**Comment by donboyd5** [comment]
No, I should. I'll get around to it, but if you talk to them first, please feel free to let them know.

**Comment by martinholmer** [comment]
@nikhilwoodruff said is issue #118:
> In https://github.com/PSLmodels/tax-microdata-benchmarking/pull/130 - E18400 in our TMD file sums to 234bn (/258bn, 10% too low) and E18500 sums to 97bn (/99bn, on target).

Apparently, the combined SALT sum of $331B (=234+97) is when using a reweighting penalty of 0.2 (the default in PR #130).

When there is no penalty (on the master branch), we get these results after the reweighting:
```
>C weighted puf SALT ($B)= 422.957 [?]
>C weighted puf SALT (#M)= 16.658 [14.3...27.1]
```

So, 423 is quite a bit more than 331.

**Comment by donboyd5** [comment]
@martinholmer, can you elaborate?

The way I read your comment is that the 0.2 penalty improved the variables we input into Tax-Calculator, which is (I think) a good thing. If I understood what @nikhilwoodruff said, our tax expenditure estimate is still far off with these improved input values, which means we need to look for other reasons for being off - including the distribution of the uncapped SALT deduction, the numbers of itemizers, which is affected by many things, and the amounts of other itemized deductions.

Is that how you interpret it?

**Comment by martinholmer** [comment]
@donboyd5 asked in issue #118:
> Is that how you interpret it?

I'm not sure; this is a complicated topic.

While the penalty weight of 0.2 may help the SALT variables, it pretty much ruins many other tax expenditures.

The one thing I feel strongly about is that when the reweighting moves things around by alot, it is likely a sign that the underlying data a not ideal.   Like we found in the QBID case with some of the business income variables being misdefined.

Looking more into the SALT-related variables themselves, and how they are constructed, seems to be a more fruitful approach than experimenting with different reweighting penalty values.

**Comment by martinholmer** [comment]
I think issue #118 has been resolved by the changes in PR #130, which were cherry-picked into the merged PR #144.

**Comment by donboyd5** [comment]
I agree. Thank you.
